### ISSUE TITLE
Late-bound Converter lookup with x:Load fails on WASM/WPF

### ISSUE THREAD

**Issue by hawkerm** [unoplatform/uno#8239]
### Current behavior

If you try adding a converter to your resources in code-behind and then use it in XAML; bindings using it fail to load properly and utilize the converter. They instead try and short-circuit and just provide the bound value as if the converter was not there.

i.e.

```cs
    Resources["ResourceValueConverter2"] = new MyConverter()
    {
        SomeValue = 10
    };
```

### Expected behavior

Converter should still be used.

### How to reproduce it (as minimally and precisely as possible)

Attached Repro Project: [UnoTestApp.NavigationBinding.zip](https://github.com/unoplatform/uno/files/8192045/UnoTestApp.NavigationBinding.zip)

1. Run the repro project on UWP
2. Note the text values specify the expected values of 5 and 10:
![image](https://user-images.githubusercontent.com/8959496/156911760-70514e92-0100-4171-8c3b-f177ade08e74.png)
4. Run the same project on WPF or WASM
5. See that the first binding is 5, but the second binding is 'FontFamily':
![image](https://user-images.githubusercontent.com/8959496/156911784-cc6f2b20-0bff-42ca-98f2-934843952e20.png)

### Workaround

This was my attempt at another work-around for #8205, so have run out of ideas. I need to have bindings evaluated with converter after other data has been loaded/set during the initialization of my page's properties.

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

4.0.11 & 4.2.0-dev.156

### Affected platforms

WebAssembly, Skia (WPF)

### IDE

Visual Studio 2022

### IDE version

17.1.0

### Anything else we need to know?

Not sure if related to #5489 or #5836 or neither.

I hit similar issues when not setting the converter in code-behind which may be related or different issues, but I couldn't easily reproduce that scenario outside my project. That was basically a cascade of properties being set including binding to a DP on a Converter set in the resources of the page, but then that value wouldn't be initialized or used when the converter was executed by the binding later in the page. In that case it did seem like the binding was firing, but the converter hadn't finished updating before the binding fired. In this case the converter seems to be being ignored entirely.

**Comment by hawkerm** [comment]
Oops, had accidently attached the repro for #8205 instead, I've updated the attachment now above to the correct repro. FYI @jeromelaban

**Comment by jeromelaban** [comment]
Implementing this feature is likely to be a bit complex at this time. As a workaround, you can set your converter in the app.xaml (even in code), at which location the converters can be referenced properly.

**Comment by hawkerm** [comment]
Yeah, in this case I'm initializing the converter with information specific to the context of the page (that's been my entire challenge on this journey 🙂); so it wouldn't work to put in the app.xaml.cs. Recalling `Bindings.Update` has helped, so I'll play more with that and see if that helps here too.

**Comment by jeromelaban** [comment]
I see, thanks. One way would be to use `x:Name` for the converter in XAML, then modify its state from the code-behind.

**Comment by MartinZikmund** [comment]
# Repro generator

This issue was classified as not reproducible with a code sample or test.

**Reason:** x:Bind/code generation bug with late-bound Converter and x:Load.
